src/parser/parsers/generics.py

```diff
@@ -0,0 +1,132 @@
+        possible_prefixes = ['m_str', 'm_map', 'm_n', 'm_fl', 'm_', 'fl', 'E', 'n']
+        parsed_generics = remove_prefixes(self.generic_data, possible_prefixes)
+
+        # Read existing generic data
+        existing_generics = self._read()
+
+        # If there ie existing data, validate the structure
+        if existing_generics is not None:
+            structure_keys_to_validate = ['ObjectiveParams', 'RejuvParams', 'ItemPricePerTier']
+            invalid_keys = validate_structures(
+                existing_generics, parsed_generics, structure_keys_to_validate
+            )
+            if len(invalid_keys) > 0:
+                print(
+                    '*WARNING* A structure within Generic data:'
+                    + f' {invalid_keys} is now different. Please verify the changes,'
+                    + ' and update the frontend page [[Module:GenericData]] accordingly.'
+                )
+
+        return parsed_generics
+
+
+def validate_structures(datas1, datas2, structure_keys_to_validate):
+    """
+    Validate that the structure (meaning shape and keys, but not value)
+    of the values for each key in datas1 and datas2 match
+
+    Returns a hash of all the invalid keys
+
+    In consecutive layers, the keys of datas1 are displayed for the invalid keys
+    """
+    invalid_keys = dict()
+    for key in structure_keys_to_validate:
+        # Ensure both contain the key
+        value1 = datas1.get(key, None)
+        value2 = datas2.get(key, None)
+        if value1 is None or value2 is None:
+            invalid_keys[key] = 'The key for this value is '
+            + 'missing in one of the data sets'
+            continue
+
+        # Check if the values differ, as this must occur first
+        if datas1[key] != datas2[key]:
+            # Ensure the types match
+            type1 = type(datas1[key])
+            type2 = type(datas2[key])
+            if type1 != type2:
+                invalid_keys[key] = 'The types of the values differ here'
+                continue
+
+            # If the value is a dictionary, recursively check the structure
+            if isinstance(datas1[key], dict):
+                more_invalid_keys = validate_structures(value1, value2, value1.keys())
+                if len(more_invalid_keys) > 0:
+                    # Add the invalid keys to the current dict
+                    invalid_keys[key] = more_invalid_keys
+
+            elif isinstance(datas1[key], list):
+                # If the value is a list, check the structure of each element that are dictionaries
+                for i, elem in enumerate(datas1[key]):
+                    if isinstance(elem, dict):
+                        more_invalid_keys = validate_structures(elem, datas2[key][i], elem.keys())
+                        if len(more_invalid_keys) > 0:
+                            invalid_keys[key] = more_invalid_keys
+
+    return invalid_keys
+
+
+def remove_prefixes(generic_data, possible_prefixes):
```

Comment:
This is probably better suited inside of the class, unless it has any use outside of it?